File: projects/ProtoMine/ProtoMine.Core/Protocol/Packets/PingRequestPacket.cs
using ProtoMine.Core.Protocol.Packets.Base;
using ProtoMine.Core.Protocol.Types;

namespace ProtoMine.Core.Protocol.Packets;

public class PingRequestPacket : Packet
{
	public override int PacketId => 0x01;

	public PingRequestPacket(int echoData)
	{
		EchoData = echoData;
	}

	[ProtocolProperty(0, typeof(ProtocolLong))]
	public long EchoData { get; }
}

File: projects/MinePack/lib/protocol/packets/v761/PingRequestPacket.ts
import { Packet } from "../Packet.js";
import { PacketProperty } from "../PacketProperty.js";

import {
	ProtocolState,
	ProtocolVersion,
	SendablePacket,
} from "../VersionedPacket.js";

import * as types from "../../../core/types/index.js";

@SendablePacket(ProtocolVersion.v761, ProtocolState.Status)
export class PingRequestPacket extends Packet {
	public readonly packetId = 0x01;

	@PacketProperty(1, types.Long)
	public payload!: number;
}
